app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -255,10 +200,75 @@ def update(self, experiment, data):
         return experiment
 
 
+class NimbusStatusValidationMixin:
+    def validate(self, data):
+        data = super().validate(data)
+
+        raised_status_field = None
+        restrictive_statuses = {
+            "status": NimbusConstants.STATUS_CHANGE_ONLY,
+            "publish_status": NimbusConstants.PUBLISH_STATUS_CHANGE_ONLY,
+        }
+
+        if self.instance:
+            status = self.instance.status
+
+            for status_field, statuses in restrictive_statuses.items():
+                instance_value = getattr(self.instance, status_field)
+                if instance_value in statuses and set(data.keys()) != {str(status_field)}:
```

Comment:
Maybe we also want the opposite here?

If the experiment status is in the list - and the set of keys proposed for update exactly match what's allowed for this status - bail out of this validator entirely because this is an allowed update?

Reply:
FWIW, I went ahead and tried this tweak locally - pytest passes for me

Reply:
A further tweak might be to just go ahead and hoist that `if raised_status_fields` conditional into that loop too.

So, like:

```diff
diff --git a/app/experimenter/experiments/api/v5/serializers.py b/app/experimenter/experiments/api/v5/serializers.py
index f9f9aa52..f97b3265 100644
--- a/app/experimenter/experiments/api/v5/serializers.py
+++ b/app/experimenter/experiments/api/v5/serializers.py
@@ -215,21 +215,20 @@ class NimbusStatusValidationMixin:
 
             for status_field, statuses in restrictive_statuses.items():
                 instance_value = getattr(self.instance, status_field)
-                if instance_value in statuses and set(data.keys()) != {str(status_field)}:
-                    raised_status_field = (status_field, instance_value)
+                if instance_value in statuses:
+                    if set(data.keys()) == {str(status_field)}:
+                        return data
+                    else:
+                        raise serializers.ValidationError(
+                            {
+                                "experiment": [
+                                    f"Nimbus Experiment has {status_field} '{instance_value}', "
+                                    f"only {status_field} can be changed."
+                                ]
+                            }
+                        )
 
-            if raised_status_field:
-                status_field = raised_status_field[0]
-                value = raised_status_field[1]
-                raise serializers.ValidationError(
-                    {
-                        "experiment": [
-                            f"Nimbus Experiment has {status_field} '{value}', "
-                            f"only {status_field} can be changed."
-                        ]
-                    }
-                )
-            elif status not in NimbusConstants.STATUS_ALLOWS_UPDATE:
+            if status not in NimbusConstants.STATUS_ALLOWS_UPDATE:
                 required_statuses = ", ".join(NimbusConstants.STATUS_ALLOWS_UPDATE)
                 raise serializers.ValidationError(
                     {
```

Reply:
Very nice, thanks for the pointer!